torch/_inductor/lowering.py

```diff
@@ -392,6 +392,9 @@ def inner_fn(index):
                 return override_fn_when_input_bool(*[load(index) for load in loaders])
             elif override_fn_when_cuda_float64 and is_cuda and dtype == torch.float64:
                 return override_fn_when_cuda_float64(*[load(index) for load in loaders])
+            elif dtype == torch.uint8 or dtype == torch.int8 or dtype == torch.int16 or dtype == torch.int32:
```

Comment:
This should be only if `not is_cuda`. A nice follow-up would be to put together tests, and if necessary, a similar fix for CUDA, if triton has C++ semantics rather than Python semantics (not sure about this).